src/contracts/centralized_oracle.js

```diff
@@ -93,12 +91,16 @@ const CentralizedOracle = {
       return;
     }
 
-    const oracle = new qweb3.Contract(contractAddress, Contracts.CentralizedOracle.abi);
-    return await oracle.call('resultSettingEndBlock', {
+    const contract = getContract(contractAddress);
+    return await contract.call('resultSettingEndBlock', {
       methodArgs: [],
       senderAddress: senderAddress,
     });
   },
 };
 
+function getContract(contractAddress) {
+  return new Contract(Config.QTUM_RPC_ADDRESS, contractAddress, Contracts.CentralizedOracle.abi);
+}
+
```

Comment:
This Contract instance as well. It's stateless

Reply:
this one cannot be a constant since I am passing in contractAddress.

Reply:
right, my mistake